velox/functions/Macros.h

```diff
@@ -66,6 +66,51 @@
   }                     \
   ;
 
+#define VELOX_UDF_BEGIN2(Name)                                                 \
+  template <typename __Velox_ExecParams>                                       \
+  struct udf {                                                                 \
```

Comment:
Ah, I had forgotten that the macros actually define a struct _within_ a struct. It'll be tricky/fragile to use inheritance to share functionality between UDFs in this framework, I agree.

Reply:
I wanted to mention this to you, but you beat me to it :-)

Reply:
@funrollloops Sagar, do you see value in changing the macros like this to make it more explicit in the code that a function is a struct?

Reply:
Making the type of the top-level thing visible is somewhat useful, but the benefit is pretty marginal IMO. On the flip side people may make the mistake I made of thinking they could inherit + override, when that won't actually work.